DynamicFeatureNavigation/XML/app/build.gradle.kts

```diff
@@ -41,55 +41,54 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
     }
 
-    dynamicFeatures = mutableSetOf(":dynamicfeature", ":includedgraphfeature")
+    dynamicFeatures.addAll(setOf(":dynamicfeature", ":includedgraphfeature"))
 
     packagingOptions {
-        exclude("META-INF/**.version")
+        resources.excludes.add("META-INF/**.version")
     }
 }
 
 dependencies {
-    api("androidx.navigation:navigation-dynamic-features-fragment:2.3.0")
-    api("androidx.appcompat:appcompat:1.1.0")
-    api("androidx.constraintlayout:constraintlayout:1.1.3")
+    api("androidx.navigation:navigation-dynamic-features-fragment:2.4.0-alpha01")
+    api("androidx.appcompat:appcompat:1.3.0")
+    api("androidx.constraintlayout:constraintlayout:2.0.4")
 }
 
+
 val bundletoolJar = project.rootDir.resolve("third_party/bundletool/bundletool-all-0.13.0.jar")
```

Comment:
we should get rid of everything below this point (it's bad practice to put it in a build file, and it uses old APIs, AND it depends on AGP tasks which is also wrong) and re-use the Gradle plugin I wrote (check if it works with newest version of AGP first) https://github.com/android/app-bundle-samples/tree/main/PlayCoreKtx/buildSrc/src/main/kotlin/com/google/android/samples/appbundle

Reply:
🔥